shopping-cart/app/shoppingcart/common/CartController.java

```diff
@@ -60,24 +62,24 @@ public CartController(final ControllerDependency controllerDependency) {
         return sphere().execute(CartCreateCommand.of(cartDraft));
     }
 
-    private F.Promise<Cart> fetchCartByIdOrNew(final String cartId, final UserContext userContext) {
+    private CompletionStage<Cart> fetchCartByIdOrNew(final String cartId, final UserContext userContext) {
         final CartByIdGet query = CartByIdGet.of(cartId)
                 .withExpansionPaths(m -> m.shippingInfo().shippingMethod());
         return sphere().execute(query)
-                .flatMap(cart -> validCartOrNew(cart, userContext));
+                .thenComposeAsync(cart -> validCartOrNew(cart, userContext), HttpExecution.defaultContext());
     }
 
-    private F.Promise<Cart> fetchCartByCustomerOrNew(final String customerId, final UserContext userContext) {
+    private CompletionStage<Cart> fetchCartByCustomerOrNew(final String customerId, final UserContext userContext) {
         final CartByCustomerIdGet query = CartByCustomerIdGet.of(customerId)
                 .withExpansionPaths(m -> m.shippingInfo().shippingMethod());
         return sphere().execute(query)
-                .flatMap(cart -> validCartOrNew(cart, userContext));
+                .thenComposeAsync(cart -> validCartOrNew(cart, userContext), HttpExecution.defaultContext());
```

Comment:
I don't want to get rid of it, I just want to think if there's a way to hide this complexity. Before you could easily read and understand the code, but now this distracts a lot.